make verify fails on Macbook with ARM64 architecture family

**How to categorize this issue?**

/area dev-productivity
/area testing
/kind bug

**What happened**:
On Macbook with:
`$(go env GOOS) = darwin`  && `$(go env GOARCH) = arm64`
when you run `make verify` then it fails with the following error log:
```
> Installing promtool
Unknown architecture
make: *** [hack/tools/bin/promtool] Error 255
```
**What you expected to happen**:
`make verify` should succeed when run on Macbook M1 chipset which belongs to `ARM64` architecture family.

**How to reproduce it (as minimally and precisely as possible)**:
Just run `make verify` on Macbook M1.

**Anything else we need to know?**:

The version of promtool used currently is `2.24.1` which does not have an archive for `darwin` && `arm64`. Version `2.26.0` onwards an archive for the same has been added. The latest version of the `promtool` is `2.34.0`

In the script there is a snippet:
```bash
case $(uname -m) in
  aarch64)
    arch="arm64"
    ;;
  x86_64)
    arch="amd64"
    ;;
  *)
    echo "Unknown architecture"
    exit -1
    ;;
esac
```
When you execute `uname -m` on macbook M1 terminal then you directly get `arm64` so it does not match any case and gives `Unknown architecture`. This also needs to be corrected.

**Environment**:
`uname -a` >> `Darwin CP2H3HJQML 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64`

- Gardener version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/hack/test-integration.sh
+++ b/hack/test-integration.sh
@@ -26,8 +26,15 @@ if ! command -v setup-envtest &> /dev/null ; then
   exit 1
 fi
 
+ARCH=
+# if using M1 macbook, use amd64 architecture build, as suggested in
+# https://github.com/kubernetes-sigs/controller-runtime/issues/1657#issuecomment-988484517
+if [[ $(uname) == 'Darwin' && $(uname -m) == 'arm64' ]]; then
+  ARCH='--arch=amd64'
+fi
+
 # --use-env allows overwriting the envtest tools path via the KUBEBUILDER_ASSETS env var just like it was before
-export KUBEBUILDER_ASSETS="$(setup-envtest use --use-env -p path ${ENVTEST_K8S_VERSION})"
+export KUBEBUILDER_ASSETS="$(setup-envtest ${ARCH} use --use-env -p path ${ENVTEST_K8S_VERSION})"
 echo "using envtest tools installed at '$KUBEBUILDER_ASSETS'"
 
 echo "> Integration Tests"

--- a/hack/tools/install-promtool.sh
+++ b/hack/tools/install-promtool.sh
@@ -21,9 +21,9 @@ echo "> Installing promtool"
 TOOLS_BIN_DIR=${TOOLS_BIN_DIR:-$(dirname $0)/bin}
 
 platform=$(uname -s | tr '[:upper:]' '[:lower:]')
-version="2.24.1"
+version="2.34.0"
 case $(uname -m) in
-  aarch64)
+  aarch64 | arm64)
     arch="arm64"
     ;;
   x86_64)

